### ISSUE TITLE
Implement ECS-Compatibility Mode

### ISSUE THREAD

**Issue by kares** [logstash-plugins/logstash-filter-metrics#60]
As a part of the effort to make plugins able to run in an ECS-Compatible manner by default in an upcoming release of Logstash, this plugin needs to either implement an ECS-Compatibility mode or certify that it does not implicitly use fields that conflict with ECS.

- in ECS mode would be nice to `event.set("event.kind","event")` for [categorization](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-kind.html#ecs-event-kind-event)?
  * maybe also an `event.source: 'logstash'` would make sense?
- otherwise having `"[{field}][mean]"` and similar - isn’t generalized in ECS

**Comment by kares** [comment]
^^^ these sound more like ECS "enhancements", please review @yaauie

**Comment by yaauie** [comment]
I agree that these are "enhancements". We may want to open a conversation about the shape of this snapshot data with the ECS team to see if they have generalizations in mind or recommendations for avoiding future collisions.

Because the metrics and timers need to be explicitly given (and are therefore not implicitly in conflict with ECS), I believe improvements we make here are largely orthogonal to the ECS-by-default effort.

**Comment by TheVastyDeep** [comment]
If this ever gets done it would be nice to fix [issue 14](https://github.com/logstash-plugins/logstash-filter-metrics/issues/14) as part of it (don't send the hostname in the [message] field).
